src/ErrorHandler.php

```diff
@@ -402,7 +409,21 @@ private function handleFatalError(array $error = null): void
             $error = error_get_last();
         }
 
-        if (!empty($error) && $error['type'] & (E_ERROR | E_PARSE | E_CORE_ERROR | E_CORE_WARNING | E_COMPILE_ERROR | E_COMPILE_WARNING)) {
+        if (empty($error)) {
+            return;
+        }
+
+        if (
+            self::$lastCapturedException
+            && self::$lastCapturedException->getFile() === $error['file']
+            && self::$lastCapturedException->getLine() === $error['line']
+            && false !== strpos($error['message'], 'Uncaught ')
+            && false !== strpos($error['message'], self::$lastCapturedException->getMessage())
```

Comment:
Those two string matches are a bit "lazy", but it's to avoid issues with different variants of the message, like:
 * `Uncaught Error: `
 * `Uncaught Exception: `
 * `Error: Uncaught Error: `
etc